test/TransitivePathTest.cpp

```diff
@@ -39,58 +44,123 @@ void assertSameUnorderedContent(const IdTable& a, const IdTable& b) {
+  makePath(IdTable input, Vars vars, TransitivePathSide left,
+           TransitivePathSide right, size_t minDist, size_t maxDist) {
+    bool useBinSearch = GetParam();
+    auto qec = getQec();
+    auto subtree = ad_utility::makeExecutionTree<ValuesForTesting>(
+        qec, std::move(input), vars);
+    return {TransitivePathBase::makeTransitivePath(
+                qec, std::move(subtree), std::move(left), std::move(right),
+                minDist, maxDist, useBinSearch),
+            qec};
+  }
+
+  [[nodiscard]] static std::shared_ptr<TransitivePathBase> makePathUnbound(
+      IdTable input, Vars vars, TransitivePathSide left,
+      TransitivePathSide right, size_t minDist, size_t maxDist) {
+    auto [T, qec] = makePath(std::move(input), vars, std::move(left),
+                             std::move(right), minDist, maxDist);
+    return T;
+  }
+
+  [[nodiscard]] static std::shared_ptr<TransitivePathBase> makePathLeftBound(
+      IdTable input, Vars vars, IdTable sideTable, size_t sideTableCol,
+      Vars sideVars, TransitivePathSide left, TransitivePathSide right,
+      size_t minDist, size_t maxDist) {
+    auto [T, qec] = makePath(std::move(input), vars, std::move(left),
+                             std::move(right), minDist, maxDist);
+    auto leftOp = ad_utility::makeExecutionTree<ValuesForTesting>(
+        qec, std::move(sideTable), sideVars);
+    return T->bindLeftSide(leftOp, sideTableCol);
+  }
+
+  [[nodiscard]] static std::shared_ptr<TransitivePathBase> makePathRightBound(
+      IdTable input, Vars vars, IdTable sideTable, size_t sideTableCol,
+      Vars sideVars, TransitivePathSide left, TransitivePathSide right,
+      size_t minDist, size_t maxDist) {
+    auto [T, qec] = makePath(std::move(input), vars, std::move(left),
+                             std::move(right), minDist, maxDist);
+    auto rightOp = ad_utility::makeExecutionTree<ValuesForTesting>(
+        qec, std::move(sideTable), sideVars);
+    return T->bindRightSide(rightOp, sideTableCol);
+  }
+};
+
+TEST_P(TransitivePathTest, idToId) {
   IdTable sub(2, makeAllocator());
   sub.push_back({V(0), V(1)});
   sub.push_back({V(1), V(2)});
   sub.push_back({V(1), V(3)});
   sub.push_back({V(2), V(3)});
 
-  IdTable result(2, makeAllocator());
-
   IdTable expected(2, makeAllocator());
   expected.push_back({V(0), V(3)});
 
   TransitivePathSide left(std::nullopt, 0, V(0), 0);
   TransitivePathSide right(std::nullopt, 1, V(3), 1);
-  TransitivePath T(getQec(), nullptr, left, right, 1,
-                   std::numeric_limits<size_t>::max());
+  auto T =
+      makePathUnbound(std::move(sub), {Variable{"?start"}, Variable{"?target"}},
+                      left, right, 1, std::numeric_limits<size_t>::max());
 
-  T.computeTransitivePath<2, 2>(&result, sub, left, right);
-  assertSameUnorderedContent(expected, result);
+  auto resultTable = T->computeResultOnlyForTesting();
+  assertSameUnorderedContent(expected, resultTable.idTable());
```

Comment:
You can get rid of the `assertSmaeUnorderedContent` and use GMock (`include <gmock/gmock.h>`
ASSERT_THAT(resultTable.idTable(), ::testing::UnorderedElementsAreArray(expected))`

It gives much better error messages (we were not aware of this when we wrote this code).